Add "Recommendations for Linux" in Tuffle Installation Instructions (#306)
trufflesuite/trufflesuite.com: src/docs/truffle/getting-started/installation.md

## Recommendations for Windows

If you're running Truffle on Windows, you may encounter some naming conflicts that could prevent Truffle from executing properly. Please see [the section on resolving naming conflicts](/docs/advanced/configuration#resolving-naming-conflicts-on-windows) for solutions.

## Recommendations for Linux

> I you're running Truffle on Linux or MacOS X, you should not install it with sudo, otherwise you might encounter some permission issues. It is recommended to install truffle as a normal user. Before we can install truffle as a normal user we have to cleanup the system. It is also recommended to install nodejs and npm with nvm (node version manager). An advantage of nvm is that you can run different node versions on the same machine.

Reviewer: ```suggestion
If you're running Truffle on Linux or macOS, you should not install it with `sudo`, otherwise you might encounter some permission issues. It is recommended to install Truffle as a normal user. Before we can install Truffle as a normal user, we have to clean-up the system. It is also recommended to install Node.js and NPM with [NVM (Node Version Manager)](https://github.com/creationix/nvm]. An advantage of NVM is that you can run different Node versions on the same machine.
```

> If you're running Truffle on Linux or macOS, you should not install it with `sudo`, otherwise you might encounter some permission issues. It is recommended to install Truffle as a normal user. Before we can install Truffle as a normal user, we have to clean-up the system. It is also recommended to install Node.js and NPM with [NVM (Node Version Manager)](https://github.com/creationix/nvm]. An advantage of NVM is that you can run different Node versions on the same machine.

Reviewer 2: At the end of the Node Version Manager link, you have a "]" and I think it should be a ")".

Author: You are right

Reviewer 3: This makes the assumption that the user has a `sudo`d install and would be confusing if this is not the case.

### Cleanup node and npm
#### Remove all packages which have been installed with `sudo npm install -g`

First we want to find out which packages have been installed with `sudo npm install -g`.
Get a list of all installed packages:

```bash

> $ sudo npm list -g --depth=0

Reviewer 2: Also, since we recommend not using `sudo`, can we also remove them from all the `npm` commands below? This would be nice as I encounter a lot of users having permissions issues when doing this.

```

```bash
/usr/lib 
|--ganache-cli@6.1.8 
|--npm@6.7.0 
+--truffle@5.0.3
```

Now we want to remove them

```bash

> $ sudo npm remove -g anache-cli

Reviewer 2: Also, is `npm remove` a real command? Shouldn't this be `uninstall`?

```

```bash
$ sudo ls -l /usr/bin/npm
lrwxrwxrwx 1 root root 38 Feb  4 08:47 /usr/bin/npm -> ../lib/node_modules/npm/bin/npm-cli.js
$ sudo rm -rf /usr/bin/npm /usr/lib/node_modules/npm/bin/npm
$ sudo rm -rf /usr/share/man/man1/npm.1.gz /usr/share/man/man1/npm.1
```

Depending on the linux distribution you might have different paths.

Now that we have cleaned up everything we can start with the installation of nvm and node.

### Installation of node and npm with nvm

#### Install nvm (node version manager)

See: https://github.com/creationix/nvm

```bash
$ curl -o- https://raw.githubusercontent.com/creationix/nvm/v0.34.0/install.sh | bash
```

In order to load the new environment, you have to close your console and open a new one.

#### Install node.js with nvm

```bash
$ nvm install node # "node" is an alias for the latest version.
```

#### Install the latest npm version

```bash
$ npm install -g npm
```

#### Delete and reset the npm prefix

See: https://stackoverflow.com/questions/34718528/nvm-is-not-compatible-with-the-npm-config-prefix-option

```bash
$ npm config delete prefix
$ npm config set prefix $NVM_DIR/versions/node/v11.10.0
```

> ### Install truffle and other packages as a normal user

Reviewer: ```suggestion
### Install Truffle and other packages as a normal user
```
